frappe/public/js/frappe/form/control.js

```diff
@@ -318,8 +318,10 @@ frappe.ui.form.ControlInput = frappe.ui.form.Control.extend({
 	},
 
 	set_disp_area: function() {
+		// to set the 0 value in readonly currency field
+		value = (this.value === 0 || this.get_value() === 0)? 0: this.value || this.get_value()
```

Comment:
this is very cryptic. Can we just have that if fieldtype is 'Currency', then it should be 0